File: fix-priority-db.py
import psycopg2
import os

DATABASE_URL = os.environ.get('DATABASE_URL', 'postgresql://postgres:postgres@localhost:5432/MTCbuildup')

conn = psycopg2.connect(DATABASE_URL)
cur = conn.cursor()

print("Updating Priority enum values in WorkOrder table...")

# Update semua data lama ke nilai baru
updates = [
    ("URGENT", "HIGH"),
    ("DARURAT", "HIGH"),
    ("HIGH", "MEDIUM"),
    ("PROSES_BERHENTI", "MEDIUM"),
    ("MEDIUM", "LOW"),
    ("LOW", "LOW"),
    ("PROSES_BERJALAN", "LOW"),
]

for old_val, new_val in updates:
    cur.execute(
        'UPDATE "WorkOrder" SET "priority" = %s WHERE "priority" = %s',
        (new_val, old_val)
    )
    print(f"  Updated {old_val} -> {new_val}")

conn.commit()
print("\nDone! Now run: npx prisma db push --accept-data-loss")
cur.close()
conn.close()

File: fix-priority-db.js
const { Pool } = require('pg');
require('dotenv').config();

const dbUrl = process.env.DATABASE_URL || 'postgresql://postgres:postgres@localhost:5432/MTCbuildup';
const pool = new Pool({ connectionString: dbUrl });

async function fixPriority() {
  const client = await pool.connect();
  
  try {
    console.log('Updating Priority enum values in WorkOrder table...');
    
    // Set semua ke LOW dulu dulu
    await client.query('UPDATE "WorkOrder" SET "priority" = \'LOW\'');
    console.log('  All set to LOW');
    
    // Hapus enum lama
    try {
      await client.query('DROP TYPE IF EXISTS "Priority_new" CASCADE');
      console.log('  Dropped Priority_new enum');
    } catch (e) { /* ignore */ }
    
    console.log('\nDone! All priorities set to LOW.');
  } catch (err) {
    console.error('Error:', err.message);
  } finally {
    client.release();
    await pool.end();
  }
}

fixPriority();
